src/zenml/integrations/lightning/orchestrators/lightning_orchestrator_entrypoint.py

```diff
@@ -0,0 +1,307 @@
+    separate STUDIO.
+
+    Raises:
+        TypeError: If the active stack's orchestrator is not an instance of
+            LightningOrchestrator.
+        ValueError: If the active stack's container registry is None.
+    """
+    # Log to the container's stdout so it can be streamed by the client.
+    logger.info("Lightning orchestrator STUDIO started.")
+
+    # Parse / extract args.
+    args = parse_args()
+
+    orchestrator_run_id = os.environ.get(
+        ENV_ZENML_LIGHTNING_ORCHESTRATOR_RUN_ID
+    )
+    if not orchestrator_run_id:
+        raise ValueError(
+            f"Environment variable '{ENV_ZENML_LIGHTNING_ORCHESTRATOR_RUN_ID}' is not set."
+        )
+
+    logger.info(f"Orchestrator run id: {orchestrator_run_id}")
+
+    deployment = Client().get_deployment(args.deployment_id)
+    filename = f"{args.run_name}.tar.gz"
+
+    pipeline_dag = {
+        step_name: step.spec.upstream_steps
+        for step_name, step in deployment.step_configurations.items()
+    }
+    entrypoint_command = StepEntrypointConfiguration.get_entrypoint_command()
+
+    active_stack = Client().active_stack
+
+    orchestrator = active_stack.orchestrator
+    if not isinstance(orchestrator, LightningOrchestrator):
+        raise TypeError(
+            "The active stack's orchestrator is not an instance of LightningOrchestrator."
+        )
+
+    # Set up credentials
+    orchestrator._set_lightning_env_vars(deployment)
+
+    pipeline_settings = cast(
+        LightningOrchestratorSettings, orchestrator.get_settings(deployment)
+    )
+
+    # Gather the requirements
+    pipeline_docker_settings = (
+        deployment.pipeline_configuration.docker_settings
+    )
+    pipeline_requirements = gather_requirements(pipeline_docker_settings)
+    pipeline_requirements_to_string = " ".join(
+        f'"{req}"' for req in pipeline_requirements
+    )
+
+    unique_resource_configs: Dict[str, str] = {}
+    main_studio_name = sanitize_studio_name(
+        f"zenml_{orchestrator_run_id}_pipeline"
+    )
+    for step_name, step in deployment.step_configurations.items():
+        step_settings = cast(
+            LightningOrchestratorSettings,
+            orchestrator.get_settings(step),
+        )
+        unique_resource_configs[step_name] = main_studio_name
+        if pipeline_settings.machine_type != step_settings.machine_type:
+            unique_resource_configs[step_name] = (
+                f"zenml-{orchestrator_run_id}_{step_name}"
+            )
+
+    logger.info(f"Creating main studio: {main_studio_name}")
+    main_studio = Studio(name=main_studio_name)
+    if pipeline_settings.machine_type:
+        main_studio.start(Machine(pipeline_settings.machine_type))
+    else:
+        main_studio.start()
+
+    logger.info("Main studio started.")
+    logger.info("Uploading code to main studio the code path: %s", filename)
+    main_studio.upload_file(
+        "/teamspace/studios/this_studio/.lightning_studio/.studiorc",
```

Comment:
I have not checked this, but there must be some overlap between this code and the code in the orchestrator (some commands that are run in the studio, ...). We should refactor this to use some common util functions in the future.